### DISCUSSION TITLE
SnakeYaml used in JMX Prometheus Exporter

### DISCUSSION THREAD

**Discussion by Boojapho** [strimzi/strimzi-kafka-operator#7321] [Q&A] [answered]
With the snakeyaml override in [this pull request](https://github.com/strimzi/strimzi-kafka-operator/pull/7258) also override the snakeyaml used in the JMX Prometheus Exporter ~~v1.7.0~~ v0.17.0?  It has the vulnerable snakeyaml (1.30) for 
[CVE-2022-38749](https://nvd.nist.gov/vuln/detail/CVE-2022-38749), [CVE-2022-38750](https://nvd.nist.gov/vuln/detail/CVE-2022-38750), and [CVE-2022-38751](https://nvd.nist.gov/vuln/detail/CVE-2022-38751).

**Comment by scholzj** [comment, marked as answer]
Can you please be more specific -> in which images or in which paths do you see the dependencies etc.

**Reply by Boojapho** [reply]
I meant v0.17.0 not v1.7.0 above.  I'll correct it.

The [pom for 3rd party libraries](https://github.com/strimzi/strimzi-kafka-operator/blob/release-0.31.x/docker-images/artifacts/kafka-thirdparty-libs/3.2.x/pom.xml#L26) points to the JMX Prometheus Agent 0.17.0.  That version uses snakeyaml 1.30 in the [collector pom](https://github.com/prometheus/jmx_exporter/blob/parent-0.17.0/collector/pom.xml#L45).  JMX Prometheus Agent 0.17.1 upgrades snakeyaml to 1.31.

**Reply by scholzj** [reply]
But there is an override added in the PR you linked above. So the question is whether you actually see it being used in come container image.

**Reply by Boojapho** [reply]
In the image, I see `snakeyaml-1.31.jar` and `jmx_prometheus_javaagent-0.17.0.jar`.  I don't see `snakeyaml-1.30.jar`.  I'm not that familiar with how Java handles its dependencies.  Is it safe to assume that if I don't see `snakeyaml-1.30.jar` in the image that the JMX Prometheus Exporter is using `snakeyaml-1.31.jar`?  Or, is it possible that it gets bundled inside of `jmx_prometheus_javaagent-0.17.0.jar`?

**Reply by scholzj** [reply]
Unless the JMX Exporter has it shaded inside its own JAR, then what matters is what JARs are in the container images, yes. In the 3rd party libs there are more dependencies using it IMHO, it should be overridden for all of them.
